test/src/subnet/main.cpp

```diff
@@ -18,45 +43,84 @@
 #include "../../../lib/vendor/include/NDTime.hpp"
 #include "../../../lib/vendor/include/iestream.hpp"
 
-#include "../../../include/subnetCadmium.hpp"
+#include "../../../include/subnet_cadmium.hpp"
 
 using namespace std;
 
 using hclock=chrono::high_resolution_clock;
 using TIME = NDTime;
 
 
-/***** SETING INPUT PORTS FOR COUPLEDs *****/
-struct input_in : public cadmium::in_port<Message_t>{};
+/**
+ *  Sets input ports for message
+ */
+
+struct input_in : public cadmium::in_port<message_t>{};
+
+/**
+ *  Sets Output ports for message
+ */
 
-/***** SETING OUTPUT PORTS FOR COUPLEDs *****/
-struct output_out: public cadmium::out_port<Message_t>{};
+struct output_out: public cadmium::out_port<message_t>{};
+
+/**
+ * \brief class ApplicationGen for Application Generator
+ *
+ *  The below class application generator(ApplicationGen) takes the file path
+ *  and transmits as message
+ */
 
-/********************************************/
-/****** APPLICATION GENERATOR *******************/
-/********************************************/
 template<typename T>
-class ApplicationGen : public iestream_input<Message_t,T> {
+class ApplicationGen : public iestream_input<message_t,T> {
     public:
+
+	/**
+	 * Default constructor for class
+	 */
+
     ApplicationGen() = default;
-    ApplicationGen(const char* file_path) : iestream_input<Message_t,
+
+    /**
+     * The below constructor of ApplicationGen class takes the input file path for
```

Comment:
It is a "parameterized constructor".
add this name of constructor.

Reply:
Added the constructor name